### ISSUE TITLE
Upload of some files triggers TypeError: argument string is required

### ISSUE THREAD

**Issue by ptasheq** [adonisjs/bodyparser#28]
Hi Guys! I'd be grateful if you'd look into this and advise.

## Package version

(Basically, it's app which recently have been created with `adonis new`)

```
"@adonisjs/ace": "^5.0.8",
"@adonisjs/auth": "^3.0.7",
"@adonisjs/bodyparser": "^2.0.5",
"@adonisjs/cors": "^1.0.7",
"@adonisjs/fold": "^4.0.9",
"@adonisjs/framework": "^5.0.9",
"@adonisjs/ignitor": "^2.0.8",
"@adonisjs/lucid": "^6.1.3",
"@adonisjs/validator": "^5.0.6",
```

## Node.js and npm version

```
$ node --version
v10.15.0
$ npm --version
6.4.1
```

## Sample Code (to reproduce the issue)

Ensure `app/config/bodyParser.js` has `files.autoProcess` set to `true` or matches upload route.

Add the following to the `start/routes.js`:
```
Route.post('/upload', async ({request, response}) => {
  return response.send();
});
```

Create any reasonable YAML file e.g.:
```
echo "a: 2" > file.yml
```

Send request via httpie and observe that app crashes
```
http -f POST http://localhost:3333/upload image@test.yml

# application logs
TypeError: argument string is required
at Object.parse (/app/node_modules/media-typer/index.js:141:11)
at new File (/app/node_modules/@adonisjs/bodyparser/src/Multipart/File.js:148:36)
at Multipart.onPart (/app/node_modules/@adonisjs/bodyparser/src/Multipart/index.js:129:26)
at Form.form.on (/app/node_modules/@adonisjs/bodyparser/src/Multipart/index.js:172:14)
at Form.emit (events.js:182:13)
at Object.cb (/app/node_modules/multiparty/index.js:642:10)
at flushEmitQueue (/app/node_modules/multiparty/index.js:626:10)
at /app/node_modules/multiparty/index.js:602:5
at handlePart (/app/node_modules/multiparty/index.js:641:3)
at Form.onParseHeadersEnd (/app/node_modules/multiparty/index.js:527:5)
at Form._write (/app/node_modules/multiparty/index.js:319:24)
at doWrite (_stream_writable.js:410:12)
at writeOrBuffer (_stream_writable.js:394:5)
at Form.Writable.write (_stream_writable.js:294:11)
at IncomingMessage.ondata (_stream_readable.js:666:20)
at IncomingMessage.emit (events.js:182:13)
```

To my mind, the problem is `multiparty` sometimes doesn't set `headers['content-type']` which causes unexpected exception to be thrown. 

After further investigation I think the problem concerns all files with mime `text/plain`:

```
$ file --mime-type test.yml 
test.yml: text/plain
$ file --mime-type test2.txt 
test2.txt: text/plain

http -f POST http://localhost:3333/upload image@test2.txt
# same result
```

**Comment by thetutlage** [comment]
Seems like `httpie` is not setting the `content-type` for the file. The curl request does work fine

```sh
curl -X POST -F "image=@file.yml;type=text/plain" http://localhost:4444/
```

There is also an issue for same with [HTTPIE](https://github.com/jakubroztocil/httpie/issues/271)

But anyways, the server should not raise an exception and should handle it gracefully. [This](https://github.com/adonisjs/adonis-bodyparser/blob/v2.0.9/src/Multipart/File.js#L147) is where the code dies, do you think, you can create a PR for same?

**Comment by ptasheq** [comment]
Hi @thetutlage,

Sure, I can create a PR. However, I'm not sure how you'd see this resolved.

The simplest (and probably not desired) solution would be adding an fallback to `application/octet-stream`. But I suppose you had something different on you mind, as it feels natural to treat it as bad request and respond with 400. Did you mean setting an error on exception like below? 

```
this._error = {}

try {
	parsedTypes = mediaTyper.parse(this.headers['content-type'])	
}
catch (e) {
	this.setError(getError('typeMissing'), 'typeMissing')	
}
```

**Comment by thetutlage** [comment]
@ptasheq The issues has been fixed with the newer version of BodyParser. However, the release targets the v5 of AdonisJS. I can look into backporting the fix to the older version, if you have no plans to upgrade to v5 right now.

Do let me know and then we can open this issue

**Comment by andreasnett** [comment]
I have this same issue for a production app. Since auth does not yet work with v5 I can't upgrade. Would be great if this can be fixed for v4.1 @thetutlage

**Comment by danpospisil** [comment]
> 
> 
> @ptasheq The issues has been fixed with the newer version of BodyParser. However, the release targets the v5 of AdonisJS. I can look into backporting the fix to the older version, if you have no plans to upgrade to v5 right now.
> 
> Do let me know and then we can open this issue

I have the same issue - backporting would be highly appriciated since its easy to kill the app (DoS attack).
